projectile.el

```diff
@@ -678,6 +679,13 @@ which we're looking."
              (setq file nil))))
     (and root (expand-file-name (file-name-as-directory root)))))
 
+(defvar-local projectile-custom-root nil
+  "Defines a custom Projectile project root.
+   This is intended to be used as a file local variable.")
+
+(defun projectile-root-file-local (dir)
```

Comment:
I wouldn't wrap this in a function at all - it should simply be the first check in the `projectile-project-root` function.